### ISSUE TITLE
FMS IO does not properly decode NF_SHORT

### ISSUE THREAD

**Issue by raphaeldussin** [NOAA-GFDL/FMS#1229]
**Describe the bug**
Netcdf variables in the "short" format are not decoded properly using scale_factor and add_offset attributes.
This leads to unphysical values for temperature and blows up the saturation vapor pressure.
```
FATAL from PE   567: compute_qs: saturation vapor pressure table overflow, nbad=    802

Bad temperature=-0.118E+05  at i=  22 pe=     0
Bad temperature=-0.118E+05  at i=  23 pe=     0
Bad temperature=-0.118E+05  at i=  24 pe=     0
```

converting the variables to single precision works but is not a viable solution for more than a small test
as file size would become unreasonably large.

**To Reproduce**
I'm using ERA5 data, properly formatted to fix time axis and increasing latitude. Files are here:
`/lustre/f2/dev/Raphael.Dussin/forcings/ERA5`

and look like this:

```
netcdf ERA5_2m_temperature_1980_2days.nc {
dimensions:
	lat = 721 ;
	lon = 1440 ;
	time = UNLIMITED ; // (48 currently)
variables:
	float lat(lat) ;
		lat:units = "degrees_north" ;
		lat:long_name = "latitude" ;
	float lon(lon) ;
		lon:units = "degrees_east" ;
		lon:long_name = "longitude" ;
	short t2m(time, lat, lon) ;
		t2m:scale_factor = 0.00194805767473842 ;
		t2m:add_offset = 260.678797394502 ;
		t2m:_FillValue = -32767s ;
		t2m:missing_value = -32767s ;
		t2m:units = "K" ;
		t2m:long_name = "2 metre temperature" ;
	float time(time) ;
		time:units = "hours since 1900-01-01 00:00:00.0" ;
		time:long_name = "time" ;
		time:calendar = "gregorian" ;

// global attributes:
		:history = "Thu May 18 11:13:15 2023: ncks -F -d time,1,48 ERA5_2m_temperature_1980.nc -o ERA5_2m_temperature_1980_2days.nc" ;
		:NCO = "netCDF Operators version 4.7.9 (Homepage = http://nco.sf.net, Code = http://github.com/nco/nco)" ;
```

and my xml is `/ncrc/home1/Raphael.Dussin/RUNS/xml/OMIP4p25_ERA5.xml`

**Expected behavior**
variables written as NF_SHORT should have scale_factor and add_offset applied to them
leading to a physical range of values.

**System Environment**
Describe the system environment, include:

gaea C5, intel-classic, 2021.5.0
`/ncrc/home2/fms/local/opt/fre-commands/bronx-20/site/ncrc5/intel-classic.mk`

**Additional context**

Fails with FMS1 and FMS2 I/O, sample run directory here:
`/lustre/f2/scratch/Raphael.Dussin/work/FMS2021.03_mom6_20230509/OM4p25_ERA5.o134513827`

**Comment by uramirez8707** [comment]
@raphaeldussin Thanks for bringing this up. FMS2_io will be updated to take into account the scale_factor and add_offset attributes.

Just to confirm, this data set if being read in via data_override?

**Comment by raphaeldussin** [comment]
that's correct! this is using data_override. Let me know when you have a fix, I'd be happy to test it.

**Comment by uramirez8707** [comment]
@raphaeldussin https://github.com/NOAA-GFDL/FMS/pull/1231 attempts to fix this. If you could help with the testing that would be great.

**Comment by uramirez8707** [comment]
ok great. thanks!

This is going to be available in the 2023.02 tag of FMS which is scheduled to be released on July 13
